### DISCUSSION TITLE
How to get node id list into column

### DISCUSSION THREAD

**Discussion by iboates** [osm2pgsql-dev/osm2pgsql#2216] [Q&A] [answered]
How can I write the `object.nodes` into a table column?

I have it defined like this:

```
local edges = osm2pgsql.define_table({
    name = 'edges_pre',
    ids = { type = 'any', id_column = 'osm_id' },
    columns = {
        { column = 'node_ids', sql_type = 'integer array' },
        { column = 'geom', type = 'linestring', projection = 4326, not_null = true },
    }
})

function osm2pgsql.process_way(object) {

    edges:insert({
        node_ids = object.nodes,
        geom = object:as_linestring()
    })

}

```

But I get:

```
2024-07-28 10:17:51  ERROR: Failed to execute Lua function 'osm2pgsql.process_way': /mnt/style/osm2pgr_default.lua:168: Error in 'insert': Invalid type 'table' for text column.

stack traceback:
        [C]: in function 'insert'
        /mnt/style/osm2pgr_default.lua:168: in function </mnt/style/osm2pgr_default.lua:150>.
```

Do I need to apply some kind of conversion?

**Comment by joto** [comment, marked as answer]
Quoting [this chapter](https://osm2pgsql.org/doc/manual.html#defining-columns) in the documentation: *For special cases you usually want to keep the type field unset and only set the sql_type field. So if you want to have an array of integer column, for instance, set only the sql_type to int[]. The type field will default to text which means you have to create the text representation of your array in the form that PostgreSQL expects it.*

See the [PostgreSQL docs](https://www.postgresql.org/docs/current/arrays.html) for that.
